analyzers/src/SonarAnalyzer.Common/Rules/Utilities/TokenTypeAnalyzerBase.cs

```diff
@@ -80,182 +95,112 @@ public IEnumerable<TokenTypeInfo.Types.TokenInfo> Spans
             }
 
-            private void ClassifyToken()
+            private void CollectClassified(TokenType tokenType, TextSpan span)
             {
-                if (IsKeyword(this.token))
+                if (string.IsNullOrWhiteSpace(token.SyntaxTree.GetText().GetSubText(span).ToString()))
                 {
-                    CollectClassified(TokenType.Keyword, this.token.Span);
                     return;
                 }
 
-                if (IsStringLiteral(this.token))
+                spans.Add(new TokenTypeInfo.Types.TokenInfo
                 {
-                    CollectClassified(TokenType.StringLiteral, this.token.Span);
-                    return;
-                }
+                    TokenType = tokenType,
+                    TextRange = GetTextRange(Location.Create(token.SyntaxTree, span).GetLineSpan())
+                });
+            }
 
-                if (IsNumericLiteral(this.token))
+            private void ClassifyToken()
+            {
+                if (IsKeyword(token))
                 {
-                    CollectClassified(TokenType.NumericLiteral, this.token.Span);
-                    return;
+                    CollectClassified(TokenType.Keyword, token.Span);
                 }
-
-                if (IsIdentifier(this.token))
+                else if (IsStringLiteral(token))
+                {
+                    CollectClassified(TokenType.StringLiteral, token.Span);
+                }
+                else if (IsNumericLiteral(token))
                 {
-                    ClassifyIdentifier(this.token, this.semanticModel);
+                    CollectClassified(TokenType.NumericLiteral, token.Span);
+                }
+                else if (IsIdentifier(token))
+                {
+                    ClassifyIdentifier(token, semanticModel);
                 }
             }
 
             private void ClassifyIdentifier(SyntaxToken token, SemanticModel semanticModel)
             {
-                var declaration = semanticModel.GetDeclaredSymbol(token.Parent);
-                if (declaration != null)
+                if (semanticModel.GetDeclaredSymbol(token.Parent) is { }  declaration)
                 {
                     ClassifyIdentifier(token, declaration);
-                    return;
                 }
-
-                var parent = GetBindableParent(token);
-                if (parent != null)
+                else if (GetBindableParent(token) is { }  parent && semanticModel.GetSymbolInfo(parent).Symbol is { } symbol)
                 {
-                    var symbol = semanticModel.GetSymbolInfo(parent).Symbol;
-                    if (symbol != null)
-                    {
-                        ClassifyIdentifier(token, symbol);
-                        return;
-                    }
-                }
-
-                if (IsContextualKeyword(token))
-                {
-                    CollectClassified(TokenType.Keyword, token.Span);
```

Comment:
It's the one mentioned in this comment: https://github.com/SonarSource/sonar-dotnet/pull/4193#discussion_r603926588

I removed it intentionally, because it was unreachable.